src/components/TodoItem.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { TodoContext } from '../TodoContext';
+import { Todo } from '../types';
+
+type Props = {
+  el: Todo,
+}
+
+export const TodoItem: React.FC<Props> = ({ el }) => {
+  const { todo, setTodo } = useContext(TodoContext);
+  const [focused, setFocused] = useState(false);
+  const [editTodo, setEditTodo] = useState(el.title);
+
+  const updateTodo = () => {
+    setTodo(todo.map((elem) => {
+      if (elem.id === el.id) {
+        return {
+          ...elem,
+          title: editTodo,
+        };
+      }
+
+      return elem;
+    }));
+  };
+
+  const handledoubleClick = () => {
+    setFocused(true);
+  };
+
+  const handleClick = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      setFocused(false);
+      updateTodo();
+    }
+
+    if (event.key === 'Escape') {
+      setEditTodo(el.title);
+      setFocused(false);
+    }
+  };
+
+  const destroyItem = (id: number) => {
+    setTodo(todo.filter(elem => elem.id !== id));
+  };
+
+  const handleInputTodo = (id: number) => {
+    setTodo(todo.map((elem) => {
+      if (elem.id === id) {
+        return {
+          ...elem,
+          completed: !elem.completed,
+        };
+      }
+
+      return elem;
+    }));
+  };
+
+  const handleOnBlur = () => {
+    setFocused(false);
+    updateTodo();
+  };
+
+  return (
+    <>
+      <li
+        key={el.id}
+        className={classNames({
+          completed: el.completed,
+          editing: focused,
+        })}
```

Comment:
li element with key should be TodoList component where you map it, you can't separate key from mapping into 2 different components